### PR TITLE
Properly check for atrribute value on redefining

### PR DESCRIPTION
**Pull request by lexmag**

### CODE REVIEW

File: lib/elixir/lib/module.ex [hunk 1]
```diff
@@ -1089,10 +1089,10 @@ defmodule Module do
         "could not call #{fun} on module #{inspect module} because it was already compiled"
   end
 
-  defp warn_if_redefining_attribute(nil, _table, _key, _value), do: false
-  defp warn_if_redefining_attribute(stack, table, key, value) do
+  defp warn_if_redefining_attribute(nil, _table, _key), do: false
+  defp warn_if_redefining_attribute(stack, table, key) do
     case :ets.lookup(table, key) do
-      [{_, v}] when not v in [nil, false, value] ->
+      [{_, val}] when val != nil and elem(val, 1) != false ->
```
Comment (josevalim) [inline lib/elixir/lib/module.ex]: Then this only applies to docs. Should we have a clause checking if key is one of the docs keys?

Comment (lexmag) [inline lib/elixir/lib/module.ex]: I've thought about this, just not sure if we're interested in `moduledoc` as well? Maybe simply `doc` checking is enough for a start, will repush.

Comment (josevalim) [inline lib/elixir/lib/module.ex]: There is for sure typedoc too. they are all listed somewhere in this file or kernel.

**Comment by lexmag** [comment]
Updated.

File: lib/elixir/lib/module.ex [hunk 2]
```diff
@@ -1089,10 +1089,12 @@ defmodule Module do
         "could not call #{fun} on module #{inspect module} because it was already compiled"
   end
 
-  defp warn_if_redefining_attribute(nil, _table, _key, _value), do: false
-  defp warn_if_redefining_attribute(stack, table, key, value) do
+  defp warn_if_redefining_attribute(nil, _table, _key), do: false
+  defp warn_if_redefining_attribute(stack, table, key) do
     case :ets.lookup(table, key) do
-      [{_, v}] when not v in [nil, false, value] ->
+      [{:doc, {_, false}}] ->
```
Comment (josevalim) [inline lib/elixir/lib/module.ex]: We also need to check for typedoc and moduledoc I think. :(

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**